x-pack/plugins/apm/server/index.ts

```diff
@@ -33,6 +36,9 @@ export function mergeConfigs(apmOssConfig: APMOSSConfig, apmConfig: APMXPackConf
     'xpack.apm.ui.maxTraceItems': apmConfig['ui.maxTraceItems'],
     'xpack.apm.ui.transactionGroupBucketSize': apmConfig['ui.transactionGroupBucketSize'],
     'xpack.apm.autocreateApmIndexPattern': apmConfig.autocreateApmIndexPattern,
+    'xpack.apm.serviceMapIndexPattern': apmConfig.serviceMapIndexPattern,
+    'xpack.apm.serviceMapDestinationIndex': apmConfig.serviceMapDestinationIndex,
```

Comment:
Who will make queries against this index? Internal kibana user, end user or a third user? If it's the internal kibana user we'll run into the same problems we did with ACM where the built-in role cannot be configured. So if someone changes it from the default service maps will stop working.

Reply:
it is used to optionally place bulk-created documents which have the connections processed by the task runner, so it would be the kibana user performing this operation on the index.

Reply:
Do you agree that this would break if the end-user uses a custom index?

Reply:
as long as they customize`serviceMapDestinationIndex` and `serviceMapIndexPattern`, to match, then it won't break the feature. We query `serviceMapIndexPattern` when we load the service maps UI, so as long as there is connection data there, it will return results.

Reply:
Afaict if the user changes `serviceMapDestinationIndex` the kibana internal user will not be able to access it. And the feature will break. This was the problems we were having with ACM.
How is this different?